neo/rawio/neuralynxrawio.py

```diff
@@ -411,6 +413,242 @@ def _rescale_event_timestamp(self, event_timestamps, dtype):
+            ts_norm = data['timestamp'] - t0_all
+
+            assert len(np.where(np.diff(data['timestamp']) < good_delta)[0]) == 0, 'this shouldnt happen'
+
+            # gap indices are detected by dt > good_delta (thus pointing to rows)
+            gap_indices = np.where(np.diff(data['timestamp']) > good_delta)[0]  # row (block) indices
+            for g in gap_indices:
+                if data['timestamp'][g+1] - data['timestamp'][g] <= max_gap_duration_to_fill * 1e6:  #, 'found a big gap, what should we do???'
+                    print(f'WARNING - big gap filled {data["timestamp"][g+1] - data["timestamp"][g]} - {ncs_info["filenames"][i]}')
+            # Get the intervals of continuous data (pointing to 'ts_norm' or the orignal (broken) data
+            data_starts = np.concatenate([np.zeros(1), gap_indices+1]).astype('int')
+            data_stops = np.concatenate([gap_indices, [data['timestamp'].size-1]]).astype('int')
+
+            # Construct bins for interpolated time axis
+            ts_bins = np.vstack([ts_interpolated, np.roll(ts_interpolated, -1)])
+            ts_bins[1, -1] = ts_bins[0, -1] + good_delta
+
+            for i0, i1 in zip(data_starts, data_stops):
+
+                # Find in which row of the interpolated time_axis the timepoint of the gap fits
+                i0_itp = np.where((ts_bins[0, :] <= ts_norm[i0]) & (ts_bins[1, :] > ts_norm[i0]))[0]
+                i1_itp = np.where((ts_bins[0, :] <= ts_norm[i1]) & (ts_bins[1, :] > ts_norm[i1]))[0]
+
+                # cant be sure enough
+                assert len(i0_itp) == 1 and len(i1_itp) == 1
+                i0_itp = i0_itp[0]
+                i1_itp = i1_itp[0]
+                assert ts_interpolated[i0_itp] <= ts_norm[i0] < ts_interpolated[i0_itp] + good_delta
+                assert ts_interpolated[i1_itp] <= ts_norm[i1] < ts_interpolated[i1_itp] + good_delta
+
+                # account for disalignment between interpolated time axis and OG time points
+                n_idx_in_block = int((ts_norm[i0] - ts_interpolated[i0_itp]) / sample_dt)
+                lin_i0_itp = sub2lin(i0_itp) + n_idx_in_block
+                lin_i1_itp = sub2lin(i1_itp) + n_idx_in_block
+
+                # write the data to the correct points!
+                records[lin_i0_itp:lin_i1_itp] = data['samples'].flatten()[sub2lin(i0): sub2lin(i1)]
+
+                # remark: I could mark the gap intervals, and interpolate the data in the gap
+
+            # Prepare the data for writing to ncs format
+            records = np.resize(records, (int(records.size/512), 512))
+            patched_data = np.core.records.fromarrays(
+                [ts_interpolated + t0_all, chan_id, sample_rate, nb_valid, records], dtype=np.dtype(ncs_dtype)
+            )
+
+            header = read_ncs_header(ncs_in_name)
+            with open(ncs_out_name, 'wb') as f:
+                f.write(header)
+                f.write(patched_data.tobytes())
+
+        print('Done!')
```

Comment:
Please remove the print statement. This will clutter the output when running a full analyis and not only the patching.